pipeline/pipelines/batch_generation_pipeline.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pipelines that run GAPIC"""
+
+import yaml
+import glob
+
+from pipeline.pipelines import code_generation_pipeline as code_gen
+from pipeline.tasks import batch_tasks
+from taskflow.patterns import linear_flow
+from pipeline.utils import config_util, pipeline_util, task_utils
+
+
+class BatchPipeline(code_gen.CodeGenerationPipelineBase):
+
+    def __init__(self, get_pipeline_tasks_func, **kwargs):
+        task_factory = BatchTaskFactory(get_pipeline_tasks_func)
+        super(BatchPipeline, self).__init__(
+            task_factory, **kwargs)
+
+    def validate_kwargs(self, **kwargs):
+        required = ['apis', 'languages', 'api_config_pattern',
+                    'artman_language_yaml', 'batch_output_dir']
+        pipeline_util.validate_exists(required, **kwargs)
+
+
+class BatchTaskFactory(code_gen.TaskFactoryBase):
+
+    def __init__(self, get_pipeline_tasks_func):
+        self.get_pipeline_tasks_func = get_pipeline_tasks_func
+        super(BatchTaskFactory, self).__init__()
+
+    def get_tasks(self, **kwargs):
+        # TODO(michaelbausor): convert to unordered flow if necessary
+        batch_flow = linear_flow.Flow('BatchFlow')
+        for single_flow in self.get_language_api_flows(**kwargs):
+            batch_flow.add(single_flow)
+
+        # TODO(michaelbausor): add task to create Github PR once staging repo
+        # is public on Github
+        final_tasks = []
+
+        flow = linear_flow.Flow('BatchGenerationFlow')
+        flow.add(batch_flow)
+        flow.add(*task_utils.instantiate_tasks(final_tasks, kwargs))
+
+        return flow
+
+    def get_language_api_flows(self, apis, languages, api_config_pattern,
+                               artman_language_yaml, **kwargs):
+        # Remove language arg from kwargs - we don't want it to override
+        # the language set in _get_configs
+        del kwargs['language']
+
+        repo_root = kwargs['repo_root']
+        api_config_yamls = _get_api_configs(api_config_pattern, apis)
+
+        for api_kwargs in _get_configs(
+                api_config_yamls, languages, artman_language_yaml, repo_root):
+
+            api_kwargs.update(kwargs)
```

Comment:
This shouldn't happen with the way this is set up. Each api_kwargs is independent of previous ones, and then values are overwritten by what is in kwargs, so kwargs is never changed and no information (i.e. settings) should leak between one api x language combination and another.